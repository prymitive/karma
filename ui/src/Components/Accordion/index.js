import React from "react";

import Collapsible from "react-collapsible";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faChevronUp } from "@fortawesome/free-solid-svg-icons/faChevronUp";
import { faChevronDown } from "@fortawesome/free-solid-svg-icons/faChevronDown";

const Trigger = ({ text, isOpen }) => (
  <div className="d-flex flex-row justify-content-between">
    <div>{text}</div>
    <div>
      <FontAwesomeIcon
        icon={isOpen ? faChevronDown : faChevronUp}
        className="text-muted"
      />
    </div>
  </div>
);

const Accordion = ({ text, content, extraProps }) => (
  <Collapsible
    triggerTagName="div"
    transitionTime={50}
    className="card"
    openedClassName="card"
    triggerClassName="card-header cursor-pointer"
    triggerOpenedClassName="card-header cursor-pointer bg-light"
    contentOuterClassName="collapse show"
    contentInnerClassName="card-body my-2"
    {...extraProps}
    trigger={<Trigger text={text} isOpen={false} />}
    triggerWhenOpen={<Trigger text={text} isOpen={true} />}
  >
    {content}
  </Collapsible>
);

export { Accordion };
