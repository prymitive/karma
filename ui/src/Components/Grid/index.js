import React from "react";
import PropTypes from "prop-types";

import { useObserver } from "mobx-react-lite";

import { AlertStore } from "Stores/AlertStore";
import { Settings } from "Stores/Settings";
import { SilenceFormStore } from "Stores/SilenceFormStore";
import { AlertGrid } from "./AlertGrid";
import { FatalError } from "./FatalError";
import { UpstreamError } from "./UpstreamError";
import { UpgradeNeeded } from "./UpgradeNeeded";
import { ReloadNeeded } from "./ReloadNeeded";
import { EmptyGrid } from "./EmptyGrid";

const Grid = ({ alertStore, settingsStore, silenceFormStore }) => {
  return useObserver(() =>
    alertStore.info.upgradeNeeded ? (
      <UpgradeNeeded newVersion={alertStore.info.version} reloadAfter={3000} />
    ) : alertStore.info.reloadNeeded ? (
      <ReloadNeeded reloadAfter={4000} />
    ) : alertStore.status.error ? (
      <FatalError message={alertStore.status.error} />
    ) : alertStore.data.upstreams.counters &&
      alertStore.data.upstreams.counters.total === 1 &&
      alertStore.data.upstreams.counters.healthy === 0 &&
      alertStore.data.upstreams.instances[0] &&
      alertStore.data.upstreams.instances[0].error !== "" ? (
      <FatalError message={alertStore.data.upstreams.instances[0].error} />
    ) : alertStore.info.version !== "unknown" &&
      alertStore.info.totalAlerts === 0 ? (
      <EmptyGrid />
    ) : (
      <React.Fragment>
        {alertStore.data.upstreams.instances
          .filter((upstream) => upstream.error !== "")
          .map((upstream) => (
            <UpstreamError
              key={upstream.name}
              name={upstream.name}
              message={upstream.error}
            />
          ))}
        <AlertGrid
          alertStore={alertStore}
          settingsStore={settingsStore}
          silenceFormStore={silenceFormStore}
        />
      </React.Fragment>
    )
  );
};
Grid.propTypes = {
  alertStore: PropTypes.instanceOf(AlertStore).isRequired,
  settingsStore: PropTypes.instanceOf(Settings).isRequired,
  silenceFormStore: PropTypes.instanceOf(SilenceFormStore).isRequired,
};

export { Grid };
