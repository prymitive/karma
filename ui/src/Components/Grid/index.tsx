import { FC } from "react";

import { observer } from "mobx-react-lite";

import { AlertStore } from "Stores/AlertStore";
import { Settings } from "Stores/Settings";
import { SilenceFormStore } from "Stores/SilenceFormStore";
import AlertGrid from "./AlertGrid";
import { FatalError } from "./FatalError";
import { UpgradeNeeded } from "./UpgradeNeeded";
import { ReloadNeeded } from "./ReloadNeeded";
import { EmptyGrid } from "./EmptyGrid";

const Grid: FC<{
  alertStore: AlertStore;
  silenceFormStore: SilenceFormStore;
  settingsStore: Settings;
}> = ({ alertStore, settingsStore, silenceFormStore }) => {
  return alertStore.info.upgradeNeeded ? (
    <UpgradeNeeded newVersion={alertStore.info.version} reloadAfter={3000} />
  ) : alertStore.info.reloadNeeded ? (
    <ReloadNeeded reloadAfter={4000} />
  ) : alertStore.status.error ? (
    <FatalError message={alertStore.status.error} />
  ) : alertStore.data.upstreams.counters &&
    alertStore.data.upstreams.counters.total === 1 &&
    alertStore.data.upstreams.counters.healthy === 0 &&
    alertStore.data.upstreams.instances[0] &&
    alertStore.data.upstreams.instances[0].error !== "" ? (
    <FatalError message={alertStore.data.upstreams.instances[0].error} />
  ) : alertStore.info.version !== "unknown" &&
    alertStore.info.totalAlerts === 0 ? (
    <EmptyGrid />
  ) : (
    <AlertGrid
      alertStore={alertStore}
      settingsStore={settingsStore}
      silenceFormStore={silenceFormStore}
    />
  );
};

export default observer(Grid);
