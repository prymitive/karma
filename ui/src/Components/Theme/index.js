import React, { useEffect } from "react";
import ReactDOM from "react-dom";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSun } from "@fortawesome/free-solid-svg-icons/faSun";

const DarkTheme = React.lazy(() => import("Styles/DarkTheme"));
const LightTheme = React.lazy(() => import("Styles/LightTheme"));

const Placeholder = () => {
  return ReactDOM.createPortal(
    <div
      style={{
        zIndex: 2000,
        backgroundColor: "#455a64",
        position: "fixed",
        top: 0,
        left: 0,
        width: "100vw",
        height: "100vh",
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        flexDirection: "column",
      }}
    >
      <FontAwesomeIcon
        icon={faSun}
        size="lg"
        spin
        style={{
          color: "#5e7a88",
          fontSize: "14rem",
        }}
      />
    </div>,
    document.body
  );
};

const ThemeContext = React.createContext();

const BodyTheme = () => {
  const context = React.useContext(ThemeContext);

  useEffect(() => {
    document.body.classList.toggle("theme-dark", context.isDark);
    document.body.classList.toggle("theme-light", !context.isDark);
  }, [context.isDark]);

  return (
    <React.Suspense fallback={<Placeholder />}>
      {context.isDark ? <DarkTheme /> : <LightTheme />}
    </React.Suspense>
  );
};

export { BodyTheme, ThemeContext };
