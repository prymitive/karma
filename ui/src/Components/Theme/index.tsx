import React, { FC, useEffect } from "react";
import ReactDOM from "react-dom";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSun } from "@fortawesome/free-solid-svg-icons/faSun";

import { Styles } from "react-select";

const DarkTheme = React.lazy(() => import("Styles/DarkTheme"));
const LightTheme = React.lazy(() => import("Styles/LightTheme"));

const Placeholder = () => {
  return ReactDOM.createPortal(
    <div
      style={{
        zIndex: 2000,
        backgroundColor: "#455a64",
        position: "fixed",
        top: 0,
        left: 0,
        width: "100vw",
        height: "100vh",
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        flexDirection: "column",
      }}
    >
      <FontAwesomeIcon
        icon={faSun}
        size="lg"
        spin
        style={{
          color: "#5e7a88",
          fontSize: "14rem",
        }}
      />
    </div>,
    document.body
  );
};

export interface ThemeCtx {
  isDark: boolean;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  reactSelectStyles: Styles<any, any>;
  animations: {
    duration: number;
  };
}

const ThemeContext = React.createContext({
  isDark: false,
  reactSelectStyles: {},
  animations: { duration: 1000 },
} as ThemeCtx);

const BodyTheme: FC = () => {
  const context = React.useContext(ThemeContext);

  useEffect(() => {
    document.body.classList.toggle("theme-dark", context.isDark);
    document.body.classList.toggle("theme-light", !context.isDark);
  }, [context.isDark]);

  return (
    <React.Suspense fallback={<Placeholder />}>
      {context.isDark ? <DarkTheme /> : <LightTheme />}
    </React.Suspense>
  );
};

export { BodyTheme, ThemeContext };
