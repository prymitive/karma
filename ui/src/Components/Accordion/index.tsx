import React, { FunctionComponent, ReactNode, useState } from "react";

import { ToggleIcon } from "Components/ToggleIcon";

const Trigger: FunctionComponent<{ text: string; isOpen: boolean }> = ({
  text,
  isOpen,
}) => (
  <div className="d-flex flex-row justify-content-between">
    <div>{text}</div>
    <div>
      <ToggleIcon isOpen={isOpen} className="text-muted" />
    </div>
  </div>
);

const Accordion: FunctionComponent<{
  text: string;
  content: ReactNode;
  defaultIsOpen?: boolean;
}> = ({ text, content, defaultIsOpen }) => {
  const [isOpen, setIsOpen] = useState(defaultIsOpen || false);

  return (
    <div className="accordion card">
      <div
        className={`card-header cursor-pointer ${isOpen ? "active" : ""}`}
        onClick={() => setIsOpen(!isOpen)}
      >
        <Trigger text={text} isOpen={isOpen} />
      </div>
      <div className={isOpen ? "card-body my-2" : ""}>{isOpen && content}</div>
    </div>
  );
};

export { Accordion };
