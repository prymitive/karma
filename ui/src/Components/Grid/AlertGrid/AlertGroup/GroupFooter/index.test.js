import React from "react";

import { mount } from "enzyme";

import toDiffableHtml from "diffable-html";

import { advanceTo, clear } from "jest-date-mock";

import {
  MockAlertGroup,
  MockAnnotation,
  MockAlert,
  MockSilence,
} from "__mocks__/Alerts";
import { MockThemeContext } from "__mocks__/Theme";
import { AlertStore } from "Stores/AlertStore";
import { SilenceFormStore } from "Stores/SilenceFormStore";
import { ThemeContext } from "Components/Theme";
import { GroupFooter } from ".";

let group;
let alertStore;
let silenceFormStore;

const MockGroup = () => {
  const group = MockAlertGroup(
    { alertname: "Fake Alert" },
    [
      MockAlert([], {}, "suppressed"),
      MockAlert([], {}, "suppressed"),
      MockAlert([], {}, "suppressed"),
    ],
    [
      MockAnnotation("summary", "This is summary", true, false),
      MockAnnotation("hidden", "This is hidden annotation", false, false),
      MockAnnotation("link", "http://link.example.com", true, true),
    ],
    { label1: "foo", label2: "bar" },
    {}
  );
  return group;
};

const MockAfterUpdate = jest.fn();

beforeEach(() => {
  alertStore = new AlertStore([]);
  silenceFormStore = new SilenceFormStore();
  group = MockGroup();
  advanceTo(new Date(Date.UTC(2000, 0, 1, 15, 0, 0)));

  alertStore.data.receivers = ["by-cluster-service", "by-name"];
});

afterEach(() => {
  jest.restoreAllMocks();
  // reset Date() to current time
  clear();
});

const MountedGroupFooter = () => {
  return mount(
    <GroupFooter
      group={group}
      alertmanagers={["default"]}
      afterUpdate={MockAfterUpdate}
      alertStore={alertStore}
      silenceFormStore={silenceFormStore}
    />,
    {
      wrappingComponent: ThemeContext.Provider,
      wrappingComponentProps: { value: MockThemeContext },
    }
  );
};

describe("<GroupFooter />", () => {
  it("matches snapshot", () => {
    const tree = MountedGroupFooter().find("GroupFooter");
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("render deduplicated silence if present", () => {
    for (const id of Object.keys(group.alerts)) {
      group.alerts[id].alertmanager[0].silencedBy = ["123456789"];
    }
    group.shared.silences = { default: ["123456789"] };
    alertStore.data.silences = {
      default: {
        "123456789": MockSilence(),
      },
    };

    const tree = MountedGroupFooter().find("GroupFooter");
    expect(tree.find("ManagedSilence")).toHaveLength(1);
  });

  it("render fallback silence if not found in alertStore", () => {
    for (const id of Object.keys(group.alerts)) {
      group.alerts[id].alertmanager[0].silencedBy = ["123456789"];
    }
    group.shared.silences = { default: ["123456789"] };
    alertStore.data.silences = {
      default: {},
    };

    const tree = MountedGroupFooter().find("GroupFooter");
    expect(tree.find("FallbackSilenceDesciption")).toHaveLength(1);
  });

  it("render fallback silence if cluster not found in alertStore", () => {
    for (const id of Object.keys(group.alerts)) {
      group.alerts[id].alertmanager[0].silencedBy = ["123456789"];
    }
    group.shared.silences = { default: ["123456789"] };
    alertStore.data.silences = {
      foo: {},
    };

    const tree = MountedGroupFooter().find("GroupFooter");
    expect(tree.find("FallbackSilenceDesciption")).toHaveLength(1);
  });

  it("mathes snapshot when silence is rendered", () => {
    for (const id of Object.keys(group.alerts)) {
      group.alerts[id].alertmanager[0].silencedBy = ["123456789"];
    }
    group.shared.silences = { default: ["123456789"] };

    alertStore.data.silences = {
      default: {
        "123456789": MockSilence(),
      },
    };
    alertStore.data.silences["default"]["123456789"].id = "123456789";

    const tree = MountedGroupFooter().find("GroupFooter");
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("renders @receiver label when alertStore.data.receivers.length > 1", () => {
    alertStore.data.receivers = ["foo", "bar"];
    const tree = MountedGroupFooter();
    expect(toDiffableHtml(tree.html())).toMatch(/@receiver:/);
  });

  it("doesn't render @receiver label when alertStore.data.receivers.length == 0", () => {
    alertStore.data.receivers = [];
    const tree = MountedGroupFooter();
    expect(toDiffableHtml(tree.html())).not.toMatch(/@receiver:/);
  });
});
