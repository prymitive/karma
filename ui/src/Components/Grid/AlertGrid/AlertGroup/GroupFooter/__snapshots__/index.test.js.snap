// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupFooter /> matches snapshot 1`] = `
"
<div class=\\"card-footer bg-card-footer-default px-2 py-1\\">
  <div class=\\"mb-1\\">
    <div class=\\"mr-1 mb-1 p-1 bg-light d-inline-block rounded components-grid-annotation text-break mw-100\\"
         data-tip=\\"Click the icon to hide annotation value\\"
    >
      <svg aria-hidden=\\"true\\"
           focusable=\\"false\\"
           data-prefix=\\"fas\\"
           data-icon=\\"search-minus\\"
           class=\\"svg-inline--fa fa-search-minus fa-w-16 mr-1 cursor-pointer\\"
           role=\\"img\\"
           xmlns=\\"http://www.w3.org/2000/svg\\"
           viewbox=\\"0 0 512 512\\"
      >
        <path fill=\\"currentColor\\"
              d=\\"M304 192v32c0 6.6-5.4 12-12 12H124c-6.6 0-12-5.4-12-12v-32c0-6.6 5.4-12 12-12h168c6.6 0 12 5.4 12 12zm201 284.7L476.7 505c-9.4 9.4-24.6 9.4-33.9 0L343 405.3c-4.5-4.5-7-10.6-7-17V372c-35.3 27.6-79.7 44-128 44C93.1 416 0 322.9 0 208S93.1 0 208 0s208 93.1 208 208c0 48.3-16.4 92.7-44 128h16.3c6.4 0 12.5 2.5 17 7l99.7 99.7c9.3 9.4 9.3 24.6 0 34zM344 208c0-75.2-60.8-136-136-136S72 132.8 72 208s60.8 136 136 136 136-60.8 136-136z\\"
        >
        </path>
      </svg>
      <span class=\\"text-muted\\">
        summary:
      </span>
      <span class=\\"Linkify\\">
        This is summary
      </span>
    </div>
    <div class=\\"mr-1 mb-1 p-1 bg-light d-inline-block rounded components-grid-annotation text-break mw-100 cursor-pointer\\"
         data-tip=\\"Click to show annotation value\\"
    >
      <svg aria-hidden=\\"true\\"
           focusable=\\"false\\"
           data-prefix=\\"fas\\"
           data-icon=\\"search-plus\\"
           class=\\"svg-inline--fa fa-search-plus fa-w-16 mr-1\\"
           role=\\"img\\"
           xmlns=\\"http://www.w3.org/2000/svg\\"
           viewbox=\\"0 0 512 512\\"
      >
        <path fill=\\"currentColor\\"
              d=\\"M304 192v32c0 6.6-5.4 12-12 12h-56v56c0 6.6-5.4 12-12 12h-32c-6.6 0-12-5.4-12-12v-56h-56c-6.6 0-12-5.4-12-12v-32c0-6.6 5.4-12 12-12h56v-56c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v56h56c6.6 0 12 5.4 12 12zm201 284.7L476.7 505c-9.4 9.4-24.6 9.4-33.9 0L343 405.3c-4.5-4.5-7-10.6-7-17V372c-35.3 27.6-79.7 44-128 44C93.1 416 0 322.9 0 208S93.1 0 208 0s208 93.1 208 208c0 48.3-16.4 92.7-44 128h16.3c6.4 0 12.5 2.5 17 7l99.7 99.7c9.3 9.4 9.3 24.6 0 34zM344 208c0-75.2-60.8-136-136-136S72 132.8 72 208s60.8 136 136 136 136-60.8 136-136z\\"
        >
        </path>
      </svg>
      hidden
    </div>
  </div>
  <span class=\\"components-label badge text-nowrap text-truncate mw-100 badge-warning components-label-dark components-label-with-hover\\"
        data-tip=\\"Click to only show alerts with this label or Alt+Click to hide them\\"
  >
    <span class=\\"components-label-name\\">
      label1:
    </span>
    <span class=\\"components-label-value\\">
      foo
    </span>
  </span>
  <span class=\\"components-label badge text-nowrap text-truncate mw-100 badge-warning components-label-dark components-label-with-hover\\"
        data-tip=\\"Click to only show alerts with this label or Alt+Click to hide them\\"
  >
    <span class=\\"components-label-name\\">
      label2:
    </span>
    <span class=\\"components-label-value\\">
      bar
    </span>
  </span>
  <span class=\\"components-label badge text-nowrap text-truncate mw-100 badge-warning components-label-dark components-label-with-hover\\"
        data-tip=\\"Click to only show alerts with this label or Alt+Click to hide them\\"
  >
    <span class=\\"components-label-name\\">
      @alertmanager:
    </span>
    <span class=\\"components-label-value\\">
      default
    </span>
  </span>
  <span class=\\"components-label badge text-nowrap text-truncate mw-100 badge-warning components-label-dark components-label-with-hover\\"
        data-tip=\\"Click to only show alerts with this label or Alt+Click to hide them\\"
  >
    <span class=\\"components-label-name\\">
      @receiver:
    </span>
    <span class=\\"components-label-value\\">
      by-name
    </span>
  </span>
  <a href=\\"http://link.example.com\\"
     target=\\"_blank\\"
     rel=\\"noopener noreferrer\\"
     class=\\"components-label-with-hover text-nowrap text-truncate badge mr-1 components-grid-annotation-link\\"
  >
    <svg aria-hidden=\\"true\\"
         focusable=\\"false\\"
         data-prefix=\\"fas\\"
         data-icon=\\"external-link-alt\\"
         class=\\"svg-inline--fa fa-external-link-alt fa-w-18 \\"
         role=\\"img\\"
         xmlns=\\"http://www.w3.org/2000/svg\\"
         viewbox=\\"0 0 576 512\\"
    >
      <path fill=\\"currentColor\\"
            d=\\"M576 24v127.984c0 21.461-25.96 31.98-40.971 16.971l-35.707-35.709-243.523 243.523c-9.373 9.373-24.568 9.373-33.941 0l-22.627-22.627c-9.373-9.373-9.373-24.569 0-33.941L442.756 76.676l-35.703-35.705C391.982 25.9 402.656 0 424.024 0H552c13.255 0 24 10.745 24 24zM407.029 270.794l-16 16A23.999 23.999 0 0 0 384 303.765V448H64V128h264a24.003 24.003 0 0 0 16.97-7.029l16-16C376.089 89.851 365.381 64 344 64H48C21.49 64 0 85.49 0 112v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V287.764c0-21.382-25.852-32.09-40.971-16.97z\\"
      >
      </path>
    </svg>
    link
  </a>
</div>
"
`;

exports[`<GroupFooter /> mathes snapshot when silence is rendered 1`] = `
"
<div class=\\"card-footer bg-card-footer-default px-2 py-1\\">
  <div class=\\"mb-1\\">
    <div class=\\"mr-1 mb-1 p-1 bg-light d-inline-block rounded components-grid-annotation text-break mw-100\\"
         data-tip=\\"Click the icon to hide annotation value\\"
    >
      <svg aria-hidden=\\"true\\"
           focusable=\\"false\\"
           data-prefix=\\"fas\\"
           data-icon=\\"search-minus\\"
           class=\\"svg-inline--fa fa-search-minus fa-w-16 mr-1 cursor-pointer\\"
           role=\\"img\\"
           xmlns=\\"http://www.w3.org/2000/svg\\"
           viewbox=\\"0 0 512 512\\"
      >
        <path fill=\\"currentColor\\"
              d=\\"M304 192v32c0 6.6-5.4 12-12 12H124c-6.6 0-12-5.4-12-12v-32c0-6.6 5.4-12 12-12h168c6.6 0 12 5.4 12 12zm201 284.7L476.7 505c-9.4 9.4-24.6 9.4-33.9 0L343 405.3c-4.5-4.5-7-10.6-7-17V372c-35.3 27.6-79.7 44-128 44C93.1 416 0 322.9 0 208S93.1 0 208 0s208 93.1 208 208c0 48.3-16.4 92.7-44 128h16.3c6.4 0 12.5 2.5 17 7l99.7 99.7c9.3 9.4 9.3 24.6 0 34zM344 208c0-75.2-60.8-136-136-136S72 132.8 72 208s60.8 136 136 136 136-60.8 136-136z\\"
        >
        </path>
      </svg>
      <span class=\\"text-muted\\">
        summary:
      </span>
      <span class=\\"Linkify\\">
        This is summary
      </span>
    </div>
    <div class=\\"mr-1 mb-1 p-1 bg-light d-inline-block rounded components-grid-annotation text-break mw-100 cursor-pointer\\"
         data-tip=\\"Click to show annotation value\\"
    >
      <svg aria-hidden=\\"true\\"
           focusable=\\"false\\"
           data-prefix=\\"fas\\"
           data-icon=\\"search-plus\\"
           class=\\"svg-inline--fa fa-search-plus fa-w-16 mr-1\\"
           role=\\"img\\"
           xmlns=\\"http://www.w3.org/2000/svg\\"
           viewbox=\\"0 0 512 512\\"
      >
        <path fill=\\"currentColor\\"
              d=\\"M304 192v32c0 6.6-5.4 12-12 12h-56v56c0 6.6-5.4 12-12 12h-32c-6.6 0-12-5.4-12-12v-56h-56c-6.6 0-12-5.4-12-12v-32c0-6.6 5.4-12 12-12h56v-56c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v56h56c6.6 0 12 5.4 12 12zm201 284.7L476.7 505c-9.4 9.4-24.6 9.4-33.9 0L343 405.3c-4.5-4.5-7-10.6-7-17V372c-35.3 27.6-79.7 44-128 44C93.1 416 0 322.9 0 208S93.1 0 208 0s208 93.1 208 208c0 48.3-16.4 92.7-44 128h16.3c6.4 0 12.5 2.5 17 7l99.7 99.7c9.3 9.4 9.3 24.6 0 34zM344 208c0-75.2-60.8-136-136-136S72 132.8 72 208s60.8 136 136 136 136-60.8 136-136z\\"
        >
        </path>
      </svg>
      hidden
    </div>
  </div>
  <span class=\\"components-label badge text-nowrap text-truncate mw-100 badge-warning components-label-dark components-label-with-hover\\"
        data-tip=\\"Click to only show alerts with this label or Alt+Click to hide them\\"
  >
    <span class=\\"components-label-name\\">
      label1:
    </span>
    <span class=\\"components-label-value\\">
      foo
    </span>
  </span>
  <span class=\\"components-label badge text-nowrap text-truncate mw-100 badge-warning components-label-dark components-label-with-hover\\"
        data-tip=\\"Click to only show alerts with this label or Alt+Click to hide them\\"
  >
    <span class=\\"components-label-name\\">
      label2:
    </span>
    <span class=\\"components-label-value\\">
      bar
    </span>
  </span>
  <span class=\\"components-label badge text-nowrap text-truncate mw-100 badge-warning components-label-dark components-label-with-hover\\"
        data-tip=\\"Click to only show alerts with this label or Alt+Click to hide them\\"
  >
    <span class=\\"components-label-name\\">
      @alertmanager:
    </span>
    <span class=\\"components-label-value\\">
      default
    </span>
  </span>
  <span class=\\"components-label badge text-nowrap text-truncate mw-100 badge-warning components-label-dark components-label-with-hover\\"
        data-tip=\\"Click to only show alerts with this label or Alt+Click to hide them\\"
  >
    <span class=\\"components-label-name\\">
      @receiver:
    </span>
    <span class=\\"components-label-value\\">
      by-name
    </span>
  </span>
  <a href=\\"http://link.example.com\\"
     target=\\"_blank\\"
     rel=\\"noopener noreferrer\\"
     class=\\"components-label-with-hover text-nowrap text-truncate badge mr-1 components-grid-annotation-link\\"
  >
    <svg aria-hidden=\\"true\\"
         focusable=\\"false\\"
         data-prefix=\\"fas\\"
         data-icon=\\"external-link-alt\\"
         class=\\"svg-inline--fa fa-external-link-alt fa-w-18 \\"
         role=\\"img\\"
         xmlns=\\"http://www.w3.org/2000/svg\\"
         viewbox=\\"0 0 576 512\\"
    >
      <path fill=\\"currentColor\\"
            d=\\"M576 24v127.984c0 21.461-25.96 31.98-40.971 16.971l-35.707-35.709-243.523 243.523c-9.373 9.373-24.568 9.373-33.941 0l-22.627-22.627c-9.373-9.373-9.373-24.569 0-33.941L442.756 76.676l-35.703-35.705C391.982 25.9 402.656 0 424.024 0H552c13.255 0 24 10.745 24 24zM407.029 270.794l-16 16A23.999 23.999 0 0 0 384 303.765V448H64V128h264a24.003 24.003 0 0 0 16.97-7.029l16-16C376.089 89.851 365.381 64 344 64H48C21.49 64 0 85.49 0 112v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V287.764c0-21.382-25.852-32.09-40.971-16.97z\\"
      >
      </path>
    </svg>
    link
  </a>
  <div class=\\"components-grid-alertgrid-alertgroup-shared-silence rounded-0 border-left-1 border-right-0 border-top-0 border-bottom-0 border-success \\">
    <div class=\\"card mt-1 border-0 p-1\\">
      <div class=\\"card-text mb-0\\">
        <span class=\\"text-muted my-1\\">
          <span width=\\"0\\">
            <span>
            </span>
            <span>
              Mocked Silence
            </span>
            <span style=\\"position: fixed; visibility: hidden; top: 0px; left: 0px;\\">
              …
            </span>
          </span>
          <span class=\\"blockquote-footer pt-1\\">
            <span class=\\"float-right cursor-pointer\\">
              <svg aria-hidden=\\"true\\"
                   focusable=\\"false\\"
                   data-prefix=\\"fas\\"
                   data-icon=\\"chevron-up\\"
                   class=\\"svg-inline--fa fa-chevron-up fa-w-14 \\"
                   role=\\"img\\"
                   xmlns=\\"http://www.w3.org/2000/svg\\"
                   viewbox=\\"0 0 448 512\\"
                   data-tip=\\"Toggle silence details\\"
              >
                <path fill=\\"currentColor\\"
                      d=\\"M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z\\"
                >
                </path>
              </svg>
            </span>
            <cite class=\\"components-grid-alertgroup-silences mr-2\\">
              me@example.com
            </cite>
            <span class=\\"badge badge-danger text-nowrap text-truncate mw-100 align-bottom\\">
              Expired
              <time datetime=\\"946688400000\\">
                14 hours ago
              </time>
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
"
`;
